site/docs/v1/tech/apis/identity-providers/_complete-login-text.adoc

```diff
@@ -0,0 +1,23 @@
+This API allows you to complete a {idp_display_name} login after authenticating a user using the {idp_display_name} API. If you are using the FusionAuth login UI with the {idp_display_name} button you will not utilize this API directly.
+
+This API is intended to be used if you want to build your own login page and you have added the {idp_display_name} login button to your own login page and you then need to complete the login with FusionAuth.
+
+ifeval::["{idp_display_name}" != "HYPR"]
+For example, if you built your own login page, you could add a "Login with {idp_display_name}" button and complete the {idp_display_name} authentication. When you complete the {idp_display_name} authentication you will have been returned {token_text_with_article} from {idp_display_name}. Using this API you can pass that {token_text} to FusionAuth and we will complete the login workflow and reconcile the user to FusionAuth.
+
+The user does not need to exist yet in FusionAuth to utilize this API, depending on the configured link:/docs/v1/tech/identity-providers/#linking-strategies[linking strategy]. The {token_text} returned from {idp_display_name} will be used to retrieve the user's email address or username and if that user does not yet exist in FusionAuth the user may be created.
+
+If [field]#createRegistration# has been enabled for this identity provider and the user does not yet have a registration for this application, a registration will be automatically created for the user. The user will be assigned any default roles configured for the application.
+
+If [field]#createRegistration# has not been enabled, a registration will not be created if one does not yet exist. This is useful if you wish to manually provision users and then subsequently allow them to login with {idp_display_name}.
+endif::[]
+
+ifeval::["{idp_display_name}" == "HYPR"]
+The user does not need to exist yet in FusionAuth to utilize this API. If a user with the email address used to initiate the request with HYPR does not yet exist in FusionAuth the user will be created.
```

Comment:
This doesn't sound right. I believe HYPR is the one IdP where the user does have to exist. Before you can call `login` for HYPR, you would have had to use the `/api/identity-provider/start` API. This API validates the user exists.